src/hotspot/share/opto/escape.cpp

```diff
@@ -732,6 +738,9 @@ void ConnectionGraph::reduce_phi(PhiNode* ophi) {
 void ConnectionGraph::verify_ram_nodes(Compile* C, Node* root) {
   Unique_Node_List ideal_nodes;
 
+  // If already failing or RAM is disabled, just return.
+  if (C->failing() || !ReduceAllocationMerges) return;
```

Comment:
1. move call to this `verify_ram_nodes()` method after `failed()` check in `compile.cpp` so you don need to check it here.
2. Check `_compile->do_reduce_allocation_merges()` instead of flag and move the check at the first line of this method.